### DISCUSSION TITLE
Mass Adding Tags

### DISCUSSION THREAD

**Discussion by m-archibald** [mealie-recipes/mealie#3652] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

When a new cookbook is created, if associated tags are not already in existence, it is a time consuming process to manually add them to each recipe. 
ex:
My family recipes - recipes tagged "my family recipes"
Wife's family recipes - recipes tagged "wife's family recipes"

we like having our recipes separated in their own cookbooks so searching through familiar recipes from our families is easy. But adding the associated tags is a time consuming process as it must be done one by one.

### Please provide a concise description of the feature that would resolve your issue.

A feature that allows for mass tag editing. by adding a tags tab to the UI, users could create or pick their desired tag, and then scroll through the recipes and click each recipe to add the desired tag. This would improve tagging efficiency substantially.

### Please consider and list out some caveats or tradeoffs made in your design decision

this may not be the best way to manage cookbook recipes, perhaps having a "add recipe to cookbook" button inside of each cookbook would be more intuitive.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by jjauzion** [comment]
Bulk edit tags / categories from the recipe browsing page would be very useful to help reorganize cookbooks and recipes.
For instance if I want to flag all my veggie recipe, it's very complicated at the moment. But If I could select several recipes and then click a "bulk edit" button to add tags it would make the whole things very easy.

Edit: If this is accepted, I'm willing to submit a PR for this feature

**Comment by michael-genson** [comment]
This can be done from the recipe data management page:
![image](https://github.com/user-attachments/assets/6601b07f-5aca-4254-8ef5-0b7c674ee3ea)

**Reply by jjauzion** [reply]
Thanks ! This is what I was looking for.
However I struggle to find this page. 
I finally found something in the doc and I was only able to access by adding the endpoint `group/data/recipes/` directly in the URL.

Is there a way to navigate to this page from the UI landing page ?

**Reply by michael-genson** [reply]
Yup, if you click on your profile in the upper left there's a few different settings pages, including the data management pages

**Reply by LunkSnee** [reply]
> This can be done from the recipe data management page: ![image](https://private-user-images.githubusercontent.com/71845777/371561635-6601b07f-5aca-4254-8ef5-0b7c674ee3ea.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njk5NzI2MzYsIm5iZiI6MTc2OTk3MjMzNiwicGF0aCI6Ii83MTg0NTc3Ny8zNzE1NjE2MzUtNjYwMWIwN2YtNWFjYS00MjU0LThlZjUtMGI3YzY3NGVlM2VhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjAxVDE4NTg1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyZDk3NjY5OTVmZjgyMWU1NzlkNDdlYTUzZjYwMGZkOWEwNDliMjdhYTlhNWZmZDRjNzVjNGZmMzUyZDliYTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7LcDorBQQMJVwn815SWp013nYo_8A4RXCXESc2chbNU)

I see this option, but it doesn't give any options to filter recipes. For example, filtering recipes that include "peanut butter" and then being able to bulk-apply a "nut allergy" tag, for example. Please consider adding "filters" to data management windows.

Thanks

**Reply by PrettyPlot** [reply]
This would work much better if on the recipe data management area, you could then filter by ingredient/tool or search term.  Then being able to do actions in bulk would be more meaningful, like if I decided at some point I want to add a "Chocolate" tag for any recipe that has cocoa powder or chocolate chips.

**Comment by mitchelbone** [comment]
Holy moly! I had no idea that this was possible and it would have been very handy many times.  Why is there no links to this page in the application's navigation menu?

**Reply by michael-genson** [reply]
Yup, if you click on your profile in the upper left there's a few different settings pages, including the data management pages

**Reply by mitchelbone** [reply]
Thank you Michael!

**Comment by Eisbergsalat** [comment]
I already categorised everything, when I found this 😆 

Well I can still Tag everything with this feature.

But for me ther is no way to mass edit utensils/tools (whatever it is in english). Is that right?
